candies/cli/parsers/standard.py

```diff
@@ -152,45 +153,70 @@ def configure_named_arg(x: inspect.Parameter, config: Dict):
                                 f'{short_prefix}{short}',
                                 **config)
 
-    def configure_flag(x: inspect.Parameter, config: Dict):
+    def configure_flag(x: inspect.Parameter, type: Type, config: Dict):
         # This actually makes an arg to behave like a flag,
         # so one could call `--x` instead of `--x y`.
         config['action'] = 'store_true'
 
         if x.default is x.empty:
             config['default'] = False
+        elif isinstance(x.default, bool):
+            config['default'] = x.default
         else:
-            config['default'] = bool(x.default)
+            raise ValueError(f"Expected a `bool` default value for "
+                             f"the flag '{x.name}' but got {x.default}.")
 
-        configure_named_arg(x, config)
+        configure_named_arg(x, type, config)
 
-    def configure_opt(x: inspect.Parameter, config: Dict):
+    def configure_opt(x: inspect.Parameter, type: Type, config: Dict):
         if x.default is not x.empty:
             config['default'] = x.default
             config['required'] = False
         else:
             config['required'] = True
 
-        configure_named_arg(x, config)
+        configure_named_arg(x, type, config)
 
     signature = inspect.signature(command.func)
 
     # Add parameters to the parser.
     for parameter in signature.parameters.values():
+        if parameter.annotation is None:
+            raise ValueError(f"Parameter '{parameter.name}' "
```

Comment:
Can we address that with a simple integration test in `arg_test.py`?

Reply:
Definitely! Will add such a test.